### ISSUE TITLE
RepaintPipeline: Preserve Original Tensor Model Output

### ISSUE THREAD

**Issue by c-merchant** [huggingface/diffusers#7545]
I am trying to use the output of the RepaintPipeline in a torch.tensor, but the inherent scaling and conversion to PIL Image or numpy makes assumptions about the original normalization used for the input model.

Add extra case for non-rescaled tensor passed to ImagePipelineOutput in repaint_pipeline.py.

I have casted between PIL, np.array, and tensor, but the lack of preservation of -1 to 0 leads to inconsistencies in rescaling.

**Comment by yiyixuxu** [comment]
> the inherent scaling and conversion to PIL Image or numpy makes assumptions about the original normalization used for the input model.

we do expect the input is normalized between -1 to 1, no?

**Comment by c-merchant** [comment]
> > the inherent scaling and conversion to PIL Image or numpy makes assumptions about the original normalization used for the input model.
> 
> we do expect the input is normalized between -1 to 1, no?

Yes, my inputs are normalized to -1 to 1, but the scaling of the image in line 222: image = (image / 2 + 0.5).clamp(0, 1) scales the output to be between 0 and 1.

**Comment by yiyixuxu** [comment]
oh, so because we normalized the input from`[0,1]` to `[-1.1]`; we also denormalize it back from `[-1,1]` to `[0,1]`
this way, our inputs and outputs have the same expected range `[0,1]`, and it is the same across all our official pipelines

This pipeline was deprecated a while ago, so we are no longer actively maintaining it. And there is indeed some inconsistency like you mentioned (it outputs `[0,1]` but allows you to pass input between `[-1,1]` directly if it is a torch tensor). For all the official pipelines, the expected value range for tensor input is `[0,1]`, regardless it is np array or tensor https://github.com/huggingface/diffusers/blob/7956c36aaa2b43584f16f7e755e46aa2d7efa721/src/diffusers/image_processor.py#L543 

If we want to update this pipeline, we would try to match the behavior of the official pipelines, would you be interested in doing this? Otherwise, feel free to make changes to your local copy. As I mentioned, this pipeline is not active ly maintained

**Comment by yiyixuxu** [comment]
closing this now since it is a deprecated pipeline
